project-addons/custom_partner/partner.py

```diff
@@ -284,9 +284,10 @@ def check_client_type(self):
 
     @api.multi
     def write(self, vals):
-        if self.parent_id.id:
-            if not vals.get('lang', False):
-                vals['lang'] = self.parent_id.lang
+        for partner in self:
+            if partner.parent_id.id:
+                if not vals.get('lang', False):
+                    vals['lang'] = partner.parent_id.lang
```

Comment:
creo que la mejor forma para hacer esto sería al fondo de todo, justo después del super:
if not vals.get('lang'):
    for partner in self:
        if partner.parent_id and partner.lang != partner.parent_id.lang:
            partner.lang = partner.parent_id.lang